sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/Segment.java

```diff
@@ -0,0 +1,91 @@
+package com.azure.storage.blob.changefeed;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.changefeed.implementation.util.BlobChangefeedCursor;
+import com.azure.storage.blob.changefeed.implementation.util.BlobChangefeedEventWrapper;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Gets events for a segment (represents approximately an hour of the changefeed).
+ */
+class Segment {
+    private static ClientLogger logger = new ClientLogger(Segment.class);
+
+    /* Changefeed container */
+    private final BlobContainerAsyncClient client;
+
+    /* Segment manifest location. */
+    private final String path;
+
+    /* Cursor associated with parent changefeed. */
+    private final BlobChangefeedCursor cfCursor;
+
+    /* User provided cursor. */
+    private final BlobChangefeedCursor userCursor;
+
+    /**
+     * Creates a segment with the associated path and cursors.
+     */
+    Segment(BlobContainerAsyncClient client, String segmentPath, BlobChangefeedCursor cfCursor,
+        BlobChangefeedCursor userCursor) {
+        this.client = client;
+        this.path = segmentPath;
+        this.cfCursor = cfCursor;
+        this.userCursor = userCursor;
+    }
+
+    /**
+     * Get all the events for the Segment.
+     * @return A reactive stream of {@link BlobChangefeedEventWrapper}
+     */
+    Flux<BlobChangefeedEventWrapper> getEvents() {
+        /* Download JSON manifest file. */
+        return client.getBlobAsyncClient(path)
+            .download().reduce(new ByteArrayOutputStream(), (os, buffer) -> {
```

Comment:
Seems like it may be worth throwing this download->reduce->map stuff in a shared method

Reply:
I think I already did that for all the ones that download a JSON file. Maybe refresh the page?